### PR TITLE
Continues removing the assumption that there is a foreground activity

### PR DESCRIPTION
**Pull request by zanderso**

### CODE REVIEW

File: packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java [hunk 1]
```diff
@@ -30,16 +30,16 @@ public static void registerWith(PluginRegistry.Registrar registrar) {
...
   }
 
-  BatteryPlugin(Context context) {
-    this.context = context;
+  BatteryPlugin(PluginRegistry.Registrar registrar) {
+    this.mRegistrar = registrar;
   }
 
-  private final Context context;
+  private final PluginRegistry.Registrar mRegistrar;
```
Comment (mravn-google) [inline packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java]: The `mMember` naming convention is not used in the existing code. I think we should either keep the current coding style, or apply the `m` prefix to all members.

Here and elsewhere.

File: packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java [hunk 2]
```diff
@@ -30,16 +30,16 @@ public static void registerWith(PluginRegistry.Registrar registrar) {
         new MethodChannel(registrar.messenger(), "plugins.flutter.io/battery");
     final EventChannel eventChannel =
         new EventChannel(registrar.messenger(), "plugins.flutter.io/charging");
-    final BatteryPlugin instance = new BatteryPlugin(registrar.context());
+    final BatteryPlugin instance = new BatteryPlugin(registrar);
```
Comment (mravn-google) [inline packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java]: AFAIK, `registrar.context()` constantly returns the same value. If that is correct, this change isn't strictly necessary. Are we doing it anyway for consistency with plugins that need the `registrar` to access the current `Activity`?

Same in other plugins that just need a `Context`, and not an `Activity`.

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java [hunk 3]
```diff
@@ -63,6 +63,12 @@ public void onMethodCall(MethodCall call, Result result) {
       return;
     }
 
+    Activity activity = mRegistrar.activity();
+    if (activity == null) {
+      result.error("NO_ACTIVITY", "image_picker plugin requires a foreground activity.", null);
```
Comment (mravn-google) [inline packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java]: You've used `"no_activity"` in all other places in this PR.

File: packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java (hunk 2 above)
Comment (zanderso) [reply packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java]: Yes, it's for consistency.

File: packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java (hunk 1 above)
Comment (zanderso) [reply packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java]: Changed to match existing style in each file.

File: packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java (hunk 3 above)
Comment (zanderso) [reply packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerPlugin.java]: Changed to no_activity here

**Pull request merged**